src/include/access/transam.h

```diff
@@ -252,6 +252,13 @@ typedef struct VariableCacheData
 	 */
 	TransactionId oldestClogXid;	/* oldest it's safe to look up in clog */
 
+	/*
+	 * These fields are also protected by OidGenLock. For tempOidStart, Shmem will
+	 * be the source of truth, as another process may have gotten there first and 
+	 * updated the start.
+	 */
```

Comment:
just fyi to keep an eye once this is merged to internal gitfarm as this struct could be different internally